src/interface.jl

```diff
@@ -51,7 +51,7 @@ function Base.getindex{T<:Integer}(b::Buffer, r::UnitRange{T})
 end
 
 Base.setindex!(b::Buffer, lines::Array, r::CappedRange) =
-    set_line_slice(b, r.start, r.stop, true, true, lines)
+    set_line_slice(b, r.start, r.stop, true, true, convert(Array{ASCIIString, 1}, lines))
```

Comment:
Ah, good catch. My testing on Julia 4.x has been somewhat lacking.
This solution will certainly work, but I wonder if it's worth littering the library with `Vector` fallbacks just to handle the case of empty arrays.